Core/Logic/routers/user_routes.py

```diff
@@ -1,9 +1,105 @@
-from fastapi import APIRouter,Request
+from fastapi import APIRouter, Request
+from pydantic.main import BaseModel
+from pydantic.types import UUID4
+from starlette.status import HTTP_200_OK, HTTP_201_CREATED, HTTP_404_NOT_FOUND
+from pydantic import EmailStr
 
 user_router = APIRouter()
 
 
+class PydanticUserModel(BaseModel):
+    password: str
+    username: str
+    first_name: str
+    last_name: str
+    email: EmailStr
+    Reg_no: str
 
-@user_router.get("/users/me")
-async def read_users_me(request: Request):
+
+class DeleteUserModel(BaseModel):
+    Reg_no: str
+
+
+@user_router.get("/users/current-user")
+async def current_user(request: Request):
     return request.user
+
+
+@user_router.get("/all-users")
+def show_all_users(request: Request):
+    from Logic.models import UserModel
+
+    if request.user.is_authenticated:
+        return list(UserModel.objects.values("Reg_no"))
+
+
+@user_router.post("/user")
+def create_user(request: Request, user: PydanticUserModel):
+    from Logic.models import UserModel
+
+    if request.user.is_authenticated and UserModel.objects.get(
+        username=request.user.username
+    ):
+        pseudouser = UserModel(
+            username=user.username,
+            first_name=user.first_name,
+            last_name=user.last_name,
+            email=user.email,
+            Reg_no=user.Reg_no,
+        )
+        pseudouser.set_password(user.password)
+        pseudouser.save()
```

Comment:
I see this part is recurring. 
Have a file called utilities and write a method for this and re-use it.